src/index.js

```diff
@@ -1,160 +1,333 @@
 'use strict'
 
-const EE = require('events').EventEmitter
-const util = require('util')
+const EventEmitter = require('events')
 const TimeCache = require('time-cache')
+const values = require('lodash.values')
+const pull = require('pull-stream')
+const lp = require('pull-length-prefixed')
+const Pushable = require('pull-pushable')
+
+const Peer = require('./peer')
 const utils = require('./utils')
 const pb = require('./message')
 const config = require('./config')
+
 const log = config.log
-const _intersection = require('lodash.intersection')
-const dialOnFloodSub = require('./dial-floodsub.js')
-const mountFloodSub = require('./mount-floodsub.js')
-const _values = require('lodash.values')
+const multicodec = config.multicodec
+const ensureArray = utils.ensureArray
 
-module.exports = PubSubGossip
+/**
+ * PubSubGossip, also known as pubsub-flood or just dumbsub,
+ * this implementation of pubsub focused on delivering an API
+ * for Publish/Subscribe, but with no CastTree Forming
+ * (it just floods the network).
+ */
+class PubSubGossip extends EventEmitter {
```

Comment:
Let's renamed the class to `FloodSub` and avoid the explanation. `PubSubGossip` was a temporary name while we agreed on the naming.